config/yaml_test.go

```diff
@@ -351,3 +351,27 @@ func TestMerge(t *testing.T) {
 		})
 	}
 }
+
+func TestNilYAMLProviderSetDefaultTagValue(t *testing.T) {
+	type Inner struct {
+		Set bool `yaml:"set" default:"true"`
```

Comment:
I think I am confused with `default` tag. My thought is `default` tag is for user to set a custom default value user like to an entity and auto apply it when that entity is not present.
The default golang behavior is set bool to `false` if entity is not present, then we don't need to have a `default` tag at all.
I think I may missed some context.